invisible_cities/detsim/simulate_S1.py

```diff
@@ -0,0 +1,194 @@
+    """
+    sol = []
+    for i in range(size):
+        P = np.random.random()
+        sol.append(brentq(S1_minimize, 0, 10000, args=P))
+    return np.array(sol)
+
+
+def generate_S1_times_from_pes(S1_pes_at_pmts : np.ndarray,
+                               hit_times      : np.ndarray)->list:
+    """Given the S1_pes_at_pmts, this function returns the times at which the pes
+    are be distributed (see generate_S1_time function).
+    It returns a list whose elements are the times at which the photoelectrons in that PMT
+    are generated.
+    Parameters:
+        :S1_pes_at_pmts: np.ndarray
+            the pes at each PMT generated by each hit
+    Returns:
+        :S1_times: list[np.ndarray,..]
+            Each element are the S1 times for a PMT. If certain sensor
+            do not see any pes, the array is empty.
+    """
+    S1_times = []
+    for pes_at_pmt in S1_pes_at_pmts:
+        repeated_times = np.repeat(hit_times, pes_at_pmt)
+        times = generate_S1_time(np.sum(pes_at_pmt))
+        S1_times.append(times + repeated_times)
+    return S1_times
+
+
+def create_S1_waveforms(S1_times      : list,
+                        buffer_length : float,
+                        bin_width     : float)->np.ndarray:
+    """
+    Create S1 waveform : S1_times are histogramed in a waveform
+    of given buffer_length and bin_width
+    Parameters
+        :S1_times: list
+            output of generate_S1_times_from_pes, list of size equal to the
+            number of pmts. Each element is an array with the times of the S1
+    Returns:
+        :wfs: np.ndarray
+            waveforms with buffer_length and bin_width
+    """
+    bins = np.arange(0, buffer_length + bin_width, bin_width)
+    wfs  = np.stack([np.histogram(times, bins=bins)[0] for times in S1_times])
+    return wfs
+
+
+def create_lighttable_function(filename : str)->Callable:
+    """ From a lighttable file, it returns a function of (x, y) for S2 signal
+    or (x, y, z) for S1 signal type. Signal type is read from the table.
+    Parameters:
+        :filename: str
+            name of the lighttable file
+    Returns:
+        :get_lt_values: Callable
+            this is a function which access the desired value inside
+            the lighttable. The lighttable values would be the nearest
+            points to the input positions. If the input positions are
+            outside the lighttable boundaries, zero is returned.
+            Input values must be vectors of same lenght, I. The output
+            shape will be (I, number_of_pmts).
+    """
+
+    lt     = pd.read_hdf(filename, "LT/LightTable")
+    Config = pd.read_hdf(filename, "LT/Config")    .set_index("parameter")
+
+    signaltype = Config.loc["signal_type"].value
+    sensor     = Config.loc["sensor"]     .value
+
+    lt = lt.set_index(["x", "y", "z"])
+    lt = lt.drop(sensor + "_total", axis=1) # drop total column
+
+    nsensors = lt.shape[-1]
+
+    if signaltype == "S1":
```

Comment:
Can't you unite this two cases, there is a lot of repeated code. You'd probably need to assign `None` to `zbins' -related arrays in case of `S2` and make `get_lt_values` be able to work with that

Reply:
A `None` type in bins does not work. The best approach I found is to include in the S2 LT a z column (let's say center of the EL gap). Then, the `get_lt_value` function would contain three arguments x, y, z in both, S1 and S2 LightTables. 

In my opinion this approach is good since we can greatly reduce the extent of this function, as well as to have symmetric LTs for S1 and S2 in the sense that they contain the same column indexes. I would go for it if does not colide with #756, that I think is not the case.